### PR TITLE
Get things working locally with the agent

### PR DESCRIPTION
**Pull request by pglass**
- Run slappy + bind in the same container
- Start named as the bind user
- Remove the mysql volume
- Run `designate-manage database` sync when the image is built
- Start designate-mdns in the mdns container (not central)
- Add in a few missing links in docker-compose.yml
- Specify the rndc key and rndc access in named.conf.options
- Use an ambassador container to link mdns and bind

### CODE REVIEW

**Comment by pglass** [comment]
I don't know if this actually works on carina, but locally you should now be able to do:

```
$ docker-compose up -d
$ make ports
API: 0.0.0.0:33060
BIND (udp): 0.0.0.0:32895
BIND (tcp): 0.0.0.0:33064

$ curl "0.0.0.0:33060/v2/zones"
...
$ <more fun stuff>
```

I was able to get all the functional tests passing.

I did remove zookeeper because I didn't need it while debugging. I'll add that back at some point.

**Comment by pglass** [comment]
This does work on carina!

File: Dockerfile [hunk 1]
```diff
@@ -43,12 +42,24 @@ RUN /etc/init.d/rabbitmq-server start && \
...
+# Copy over our designate config
+COPY etc/designate /etc/designate
+
+# add a separate designate.conf with localhost for the db connection so we can
+# use designate-manage to sync the db even if `mysql` is not in /etc/hosts
+RUN sed "s/@mysql/@127.0.0.1/g" /etc/designate/designate.conf > /etc/designate/local-mysql.conf
+
 # NOTE: A new password can be included in the setup_databases.sql
-RUN /etc/init.d/mysql start && mysql -u root --password=password < setup_databases.sql
+RUN /etc/init.d/mysql start && mysql -u root --password=password < setup_databases.sql && designate-manage --config-file /etc/designate/local-mysql.conf database sync
```
Comment (filler) [inline Dockerfile]: Minor nit, but perhaps line continuation mark these for readability?

```
RUN /etc/init.d/mysql start && \
  mysql -u root --password=password < setup_databases.sql && \
  designate-manage --config-file /etc/designate/local-mysql.conf database sync
```

**Comment by filler** [comment]
This is too cool for school, @pglass.  Nice work.

**Comment by pglass** [comment]
@filler done!

**Comment by filler** [comment]
:rocket: :badoosh:

**Pull request merged**